config/configauth/mock_clientauth.go

```diff
@@ -15,51 +15,7 @@
 package configauth // import "go.opentelemetry.io/collector/config/configauth"
 
 import (
-	"context"
-	"errors"
-	"net/http"
-
-	"google.golang.org/grpc/credentials"
-
-	"go.opentelemetry.io/collector/component"
-)
-
-var (
-	_            ClientAuthenticator = (*MockClientAuthenticator)(nil)
-	errMockError                     = errors.New("mock Error")
+	"go.opentelemetry.io/collector/extension/auth"
 )
 
-// MockClientAuthenticator provides a mock implementation of GRPCClientAuthenticator and HTTPClientAuthenticator interfaces
-type MockClientAuthenticator struct {
-	ResultRoundTripper      http.RoundTripper
-	ResultPerRPCCredentials credentials.PerRPCCredentials
-	MustError               bool
-}
-
-// Start for the MockClientAuthenticator does nothing
-func (m *MockClientAuthenticator) Start(ctx context.Context, host component.Host) error {
-	return nil
-}
-
-// Shutdown for the MockClientAuthenticator does nothing
-func (m *MockClientAuthenticator) Shutdown(ctx context.Context) error {
-	return nil
-}
-
-// RoundTripper for the MockClientAuthenticator either returns error if the mock authenticator is forced to or
-// returns the supplied resultRoundTripper.
-func (m *MockClientAuthenticator) RoundTripper(base http.RoundTripper) (http.RoundTripper, error) {
-	if m.MustError {
-		return nil, errMockError
-	}
-	return m.ResultRoundTripper, nil
-}
-
-// PerRPCCredentials for the MockClientAuthenticator either returns error if the mock authenticator is forced to or
-// returns the supplied resultPerRPCCredentials.
-func (m *MockClientAuthenticator) PerRPCCredentials() (credentials.PerRPCCredentials, error) {
-	if m.MustError {
-		return nil, errMockError
-	}
-	return m.ResultPerRPCCredentials, nil
-}
+type MockClientAuthenticator = auth.MockClientAuthenticator
```

Comment:
Now that we are moving things around, how about moving the MockClientAuthenticator to `authtest` instead to `auth`?